libs/domains/services/data-access/src/lib/domains-services-data-access.ts

```diff
@@ -384,6 +424,40 @@ export const mutations = {
     const response = await mutation(serviceId, deploymentRestrictionId)
     return response.data
   },
+  async editService({ serviceId, payload }: ServiceRequest) {
+    const response = await match(payload.serviceType)
+      .with(
+        'APPLICATION',
+        async () =>
```

Comment:
`async` / `await` aren't necessary if you bind the function with parameter then use it to extract `data`.
eg. `deleteDeploymentRestriction`

Reply:
You should write it like that:
```ts
    const mutation = await match(payload)
      .with(
        { serviceType: 'APPLICATION' },
        (payload) => applicationMainCallsApi.editApplication.bind(applicationMainCallsApi, serviceId, refactoApplication(payload))
      )
      .with(
        { serviceType: 'CONTAINER' },
        (payload) => containerMainCallsApi.editContainer.bind(containerMainCallsApi, serviceId, refactoContainer(payload))
      )
      .with(
        { serviceType: 'DATABASE' },
        (payload) => databaseMainCallsApi.editDatabase.bind(databaseMainCallsApi, serviceId, refactoDatabase(payload))
      )
      .with(
        { serviceType: 'JOB' },
        (payload) => jobMainCallsApi.editJob.bind(jobMainCallsApi, serviceId, refactoJob(payload))
      )
      .with(
        { serviceType: 'HELM' },
        (payload) => helmMainCallsApi.editHelm.bind(helmMainCallsApi, serviceId, refactoHelm(payload))
      )
      .exhaustive()
    const response = await mutation()
    return response.data

```